statsmodels/stats/multicollinearity.py

```diff
@@ -0,0 +1,460 @@
+        self.k_vars = triu.shape[1]
+        self.triu2 = triu**2
+
+    @cache_readonly
+    def vif(self):
+        """Variance inflation factor.
+        """
+        return self.tss / self.rss
+
+    @property
+    def tss(self):
+        """Total sum of squares of squares in sequence of regression"""
+        return self.triu2.sum(0)
+
+    @property
+    def rss(self):
+        """Residual sum of squares of squares in sequence of regression"""
+        return np.diag(self.triu2)
+
+    @property
+    def partial_corr(self):
+        """Partial correlation of one variable with all previous variables
+
+        This corresponds to R^2 in a linear regression of one variable on all other
+        variables that are before in the sequence, including a constant if standarize
+        is true.
+
+        The interpretation as correlation assumes that standardize is true.
+
+        not cached
+
+        TODO: we might want to use name `partial_corr` for partial correlation of pairs
+        given all others.
+        """
+        return 1 - self.rss / self.tss
+
+
+def vif(data, standardize=True, moment_matrix=None):
+    """Variance inflation factor
+
+    The standard interpretation requires standardize is true, or that the data is already
+    standardized or that a given moment_matrix is a correlation matrix.
+
+    Parameters
+    ----------
+    data : array_like
+        data with observations in rows and variables in columns. This is ignored if
+        moment_matrix is not None. Data can be of smaller than full rank.
+    moment_matrix : None or ndarray
+        Optional moment or correlation matrix to be used for the analysis. If it is provided
+        then data is ignored.
+        The moment_matrix needs to be nonsingular because Cholesky decomposition is used.
+    standardize : bool, default True
+        If true and data are provided, then the data will be standardized to mean zero and
+        standard deviation equal to one, which is equivalent to using the correlation matrix.
+        TODO: This might be replaced by separate demean option. Variance inflation factor
+        can be calculated for data that is scaled but not demeaned.
+        See Notes in class MultiCollinear
+
+    Return
+    ------
+    vif : ndarray or pandas.Series
+        variance inflation factor. This will be a pandas Series if the data has a `columns`
+        attribute as provided by a pandas DataFrame.
+
+    """
+    # most parts are duplicate code, copied from vif_selection
+    x = np.asarray(data)
+    # TODO: use pandas corrcoef below to have nan handling ?
+
+    if moment_matrix is not None:
+        xm = np.asarray(moment_matrix)
+        if standardize:
+            # check if we have correlation matrix,
+            # we cannot demean but we can scale
+            # (We could demean if const is included in uncentred moment matrix)
+            if (np.diag(xm) != 1).any():
+                xstd = np.sqrt(np.diag(xm))
```

Comment:
this part is not covered in unit tests

this might be wrong if moment matrix is not divided by nobs `np.diag(xm) != 1` or `np.diag(xm) != nobs` ?